From 04e823ab71c6303f31001d7eaf5bccdef50c706a Mon Sep 17 00:00:00 2001
From: Keith Mok <kmok@cyngn.com>
Date: Fri, 4 Dec 2015 10:03:49 -0800
Subject: [PATCH 5/8] keymaster: add TARGET_PROVIDES_KEYMASTER

Qcom makes keymaster a propertiies blob in M.
To maintain compatibility with Cyanogenmod builds,
add a flag to optionally disable building keymaster here.

Change-Id: Ibb95cb54651ec5af266504e66f1078adaa53c0d8
---
 Android.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Android.mk b/Android.mk
index f0e1a21..b331fb0 100644
--- a/Android.mk
+++ b/Android.mk
@@ -2,6 +2,7 @@
 ifneq ($(BOARD_IS_AUTOMOTIVE),true)
 LOCAL_PATH := $(call my-dir)
 
+ifneq ($(TARGET_PROVIDES_KEYMASTER),true)
 ifneq ($(filter msm8960 msm8226 msm8974 msm8610 msm8084 apq8084 msm8909 msm8916 msm8952 msm8992 msm8994,$(TARGET_BOARD_PLATFORM)),)
 
 keymaster-def := -fvisibility=hidden -Wall
@@ -45,4 +46,5 @@ LOCAL_MODULE_TAGS := optional
 include $(BUILD_SHARED_LIBRARY)
 
 endif # TARGET_BOARD_PLATFORM
+endif # TARGET_PROVIDES_KEYMASTER
 endif
-- 
2.17.1

